src/nvim/eval/typval.c

```diff
@@ -439,33 +502,32 @@ list_T *tv_list_copy(const vimconv_T *const conv, list_T *const orig,
   }
 
   list_T *copy = tv_list_alloc();
+  tv_list_ref(copy);
   if (copyID != 0) {
     // Do this before adding the items, because one of the items may
     // refer back to this list.
     orig->lv_copyID = copyID;
     orig->lv_copylist = copy;
   }
-  listitem_T *item;
-  for (item = orig->lv_first; item != NULL && !got_int;
-       item = item->li_next) {
+  TV_LIST_ITER(orig, item, {
```

Comment:
Missing `got_int` check. Was this on purpose?

Reply:
It appears that I can have got_int check here, but it is not exactly useful:

```
local helpers = require('test.functional.helpers')(after_each)

local command = helpers.command
local meths = helpers.meths
local clear = helpers.clear
local funcs = helpers.funcs
local sleep = helpers.sleep
local wait = helpers.wait
local feed = helpers.feed
local eq = helpers.eq

local screen
local dur
local min_dur = 8
local len = 131072

describe('List support code', function()
  before_each(function()
    clear()
    if dur == nil then
      dur = 0
      while true do
        command(([[
          let rt = reltime()
          let bl = range(%u)
          let dur = reltimestr(reltime(rt))
        ]]):format(len))
        dur = tonumber(meths.get_var('dur'))
        if dur >= min_dur then
          print(('Using len %u, dur %g'):format(len, dur))
          break
        else
          len = len * 2
        end
      end
    end
  end)
  it('allows interrupting copy', function()
    feed(':let t_rt = reltime()<CR>:let t_bl = copy(bl)<CR>')
    sleep(min_dur / 16 * 1000)
    feed('<C-c>')
    wait()
    command('let t_dur = reltimestr(reltime(t_rt))')
    local t_dur = tonumber(meths.get_var('t_dur'))
    if t_dur >= dur / 8 then
      eq(nil, ('Took too long to cancel: %g >= %g'):format(t_dur, dur / 8))
    end
  end)
end)
```

with got_int check added always fails. Manual testing confirms that copy() accually runs *slower*  when cancelled because the only thing it does cancel is an assignment, which ultimately means that big list needs to be freed before continuing.

Test succeeds if I not only add `if (got_int) break;` but also `os_breakcheck()`.

Reply:
Though I should say removal was not *on purpose*.